larpandoracontent/LArThreeDReco/LArTwoViewMatching/TwoViewThreeDKinkTool.cc

```diff
@@ -0,0 +1,480 @@
+
+                float transverseAB(std::numeric_limits<float>::max()), transverseBA(std::numeric_limits<float>::max());
+                float longitudinalAB(-std::numeric_limits<float>::max()), longitudinalBA(-std::numeric_limits<float>::max());
+
+                LArPointingClusterHelper::GetImpactParameters(vertexA, vertexB, longitudinalAB, transverseAB);
+                LArPointingClusterHelper::GetImpactParameters(vertexB, vertexA, longitudinalBA, transverseBA);
+
+                if (std::min(longitudinalAB, longitudinalBA) < m_minLongitudinalImpactParameter)
+                    continue;
+
+                if (std::min(transverseAB, transverseBA) > m_maxTransverseImpactParameter)
+                    continue;
+
+                const float cosTheta(-vertexA.GetDirection().GetCosOpeningAngle(vertexB.GetDirection()));
+
+                if (cosTheta < m_minImpactParameterCosTheta)
+                    continue;
+
+                const bool isALowestInX(this->IsALowestInX(pointingClusterA, pointingClusterB));
+                const CartesianVector splitPosition((vertexA.GetPosition() + vertexB.GetPosition()) * 0.5f);
+                const bool isThreeDKink(this->IsThreeDKink(pAlgorithm, particle, splitPosition, isALowestInX));
+
+                if (isThreeDKink != m_splitMode)
+                    continue;
+
+                // Construct the modification object
+                Modification modification;
+
+                if (m_splitMode)
+                {
+                    const TwoDSlidingFitResult &fitResultCommon(pAlgorithm->GetCachedSlidingFitResult(particle.m_pCommonCluster));
+
+                    CartesianVector splitPositionCommon(0.f, 0.f, 0.f);
+                    if (STATUS_CODE_SUCCESS != fitResultCommon.GetGlobalFitPositionAtX(splitPosition.GetX(), splitPositionCommon))
+                    {
+                        continue;
+                    }
+
+                    modification.m_splitPositionMap[particle.m_pCommonCluster].push_back(splitPositionCommon);
+                }
+                else
+                {
+                    const bool vertexAIsLowX(vertexA.GetPosition().GetX() < vertexB.GetPosition().GetX());
+                    const Cluster *const pLowXCluster(vertexAIsLowX ? particle.m_pClusterA : particle.m_pClusterB);
+                    const Cluster *const pHighXCluster(vertexAIsLowX ? particle.m_pClusterB : particle.m_pClusterA);
+                    modification.m_clusterMergeMap[pLowXCluster].push_back(pHighXCluster);
+                }
+
+                modification.m_affectedClusters.push_back(particle.m_pClusterA);
+                modification.m_affectedClusters.push_back(particle.m_pClusterB);
+                modification.m_affectedClusters.push_back(particle.m_pCommonCluster);
+
+                modificationList.push_back(modification);
+            }
+            catch (StatusCodeException &statusCodeException)
```

Comment:
This can be const, although suspect it isn't (yet) in three-view equivalent!

Reply:
That's the case. Should I change it there as well, or leave it for a later revision?

Reply:
Please feel free to change this wherever it's applicable. Thanks!